src/components/Review/ReviewSectionRowAssigned.tsx

```diff
@@ -1,72 +1,93 @@
 import React from 'react'
 import { Grid, Icon, Label } from 'semantic-ui-react'
 import strings from '../../utils/constants'
+import { User } from '../../utils/generated/graphql'
 import { ReviewAction, ReviewSectionComponentProps } from '../../utils/types'
 
 const ReviewSectionRowAssigned: React.FC<ReviewSectionComponentProps> = ({
   isAssignedToCurrentUser,
   action,
   assignment,
 }) => {
-  const isSelfAssigned: boolean =
-    action === ReviewAction.canStartReview || action === ReviewAction.canContinue
-  const isReviewLocked: boolean =
-    action === ReviewAction.canSelfAssignLocked || action === ReviewAction.canContinueLocked
-  return (
-    <Grid.Column className="padding-zero">
-      {isAssignedToCurrentUser && action === ReviewAction.canSelfAssign ? (
-        <Label className="simple-label" content={strings.LABEL_ASSIGNMENT_SELF} />
-      ) : isSelfAssigned ? (
-        <Label
-          className="simple-label relevant-text"
-          sie="large"
-          icon={<Icon name="circle" size="tiny" color="blue" />}
-          content={strings.LABEL_ASSIGNED_TO_YOU}
-        />
-      ) : isReviewLocked ? (
-        <Label
-          className="simple-label"
-          icon={<Icon name="ban" size="small" color="red" />}
-          content={
-            <>
-              {`${strings.LABEL_ASSIGNMENT_LOCKED} `}
-              <Label
-                className="simple-label"
-                content={
-                  isAssignedToCurrentUser
-                    ? strings.REVIEW_FILTER_YOURSELF
-                    : `${assignment.reviewer?.firstName || ''} ${
-                        assignment.reviewer?.lastName || ''
-                      }`
-                }
-              />
-            </>
-          }
-        />
-      ) : (
-        <Label
-          className="simple-label"
-          content={
-            <>
-              {action === ReviewAction.canSelfAssign
-                ? `${strings.LABEL_ASSIGNMENT_AVAILABLE} `
-                : `${strings.LABEL_REVIEWED_BY} `}
-              <Label
-                className="simple-label info-text"
-                content={
-                  isAssignedToCurrentUser
-                    ? strings.REVIEW_FILTER_YOURSELF
-                    : `${assignment.reviewer?.firstName || ''} ${
-                        assignment.reviewer?.lastName || ''
-                      }`
-                }
-              />
-            </>
-          }
-        />
-      )}
-    </Grid.Column>
-  )
+  const getLabel = () => {
+    switch (action) {
+      case ReviewAction.canSelfAssign:
+        return isAssignedToCurrentUser ? (
+          <CurrentSelfAssignmentLabel />
+        ) : (
+          <TheirSelfAssignmentLabel {...assignment.reviewer} />
+        )
+      case ReviewAction.canStartReview:
```

Comment:
should this also include `canReReview` and `canUpdate` ?

Reply:
Yes, sure thing